### PR TITLE
Night Elf Facial Markings

### PR DESCRIPTION
**Pull request by ercarp**
## Changelog:
- Added **8** facial markings for the Night Elf ethnicity based on the ones used in _World of Warcraft_.
  - _One marking ("Shadow") is still missing. This will be added later._
- The markings are used by Night Elf females with the **_Elunic_** religion. There is a small chance for the game to spawn in characters without markings, but the majority (98%) of Night Elves have them.
  - Additionally, characters with the **_Zealous_** trait are guaranteed to always have markings.
- Markings base their color off the character's hair color.
- Markings have been added to Ruler Designer and Barbershop.

<details><summary>Click to view image</summary>

image

</details>

### To-do:
- An event or decision for acquiring facial markings as a rite of adulthood. This will be done on a different branch, however.
- [x] Add them in the Ruler Designer.
- [x] Unite gene_facial_markings and gene_scripted_makeup
- [x] Localized this:
<details>
<summary>Click to expand</summary>

image

</details>

- [x] Localized special markings

# How to test:
1. Provide feedback in the comments and if needed, make suggestions how you think they could be improved visually.

### CODE REVIEW

**Review by Deuce20201 — CHANGES_REQUESTED** [review]
Markings look good. 

But 98% of females generate the marking but not Tyrande Whisperwind. 



Tyrande_Whisperwind

**Comment by ercarp** [comment]
> But 98% of females generate the marking but not Tyrande Whisperwind.

Tyrande will likely have custom DNA eventually with unique markings, so this isn't that big of a deal.

**Comment by Deuce20201** [comment]
> > But 98% of females generate the marking but not Tyrande Whisperwind.
> 
> Tyrande will likely have custom DNA eventually with unique markings, so this isn't that big of a deal.

In that case I have no other criticisms.

**Review by Deuce20201 — APPROVED** [review]

File: gfx/portraits/portrait_modifiers/wc_facial_markings.txt [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+				template = no_markings
+				range = { 1 1 }
+			}
+		}
+		weight = {
+			base = 0
+			modifier = {
+				add = 2
+				has_elunes_marks_trigger = yes
+				NOT = { has_trait = zealous }
```
Comment (zumbak04) [inline gfx/portraits/portrait_modifiers/wc_facial_markings.txt:19]: I would make it so:
Only those who have the Cynical trait don't have markings.

**Review by TheGroxEmpire — COMMENTED** [review]
Some bright hair colors from generated character (ruler designer and in game) are giving really bright color markings that are imo weird:
<details>

image
image
image
image

</details>

**Comment by TheDrPibb1999** [comment]
got the same weird color as grox

**Comment by ercarp** [comment]
@TheGroxEmpire @TheDrPibb1999 This has to do with the blend mode used for the markings in `common/genes/wc_gene_facial_markings.txt`. There are four blend modes to choose from: `overlay`, `replace`, `hard_light`, and `multiply`. The current one being used is `overlay`, and there is a reason for it which I will try to illustrate.

**Overlay** seems to work best with mid-range colors. **Multiply** with dark and mid-range colors (lighter colors are very faded, some of them don't show at all). **Hard Light** and **Replace** function very similarly.

If you want the colors to be exact, then Replace would probably be the best option for a blend mode, but it can also look a bit too much like paint. I suggest you navigate to the file mentioned above and switch around the blend modes to see which ones you personally think look best.

<details><summary>Click to expand</summary>

image

</details>

**Review by TheGroxEmpire — COMMENTED** [review]
Thank you for the detailed explanation. Would it be possible to limit the range that it the marking color may be generated? Color like white, light pink, etc might not just be suitable for markings color.

Most of the markings are pretty good given the right color. My next suggestion will be to increase the opacity on them. The base opacity is around half slider in ruler designer:
<details>

Base opacity:
image
image
image


Increased opacity:
image
image
image
</details>

One more thing, I found that the dark green haired Nelves have almost black markings, this is pretty unusual for me. It could be a bit more green colored imo:
<details>

image
</details>

**Comment by TheDrPibb1999** [comment]
Personally, IMO i disagree with "One more thing, I found that the dark green haired Nelves have almost black markings, this is pretty unusual for me. It could be a bit more green-colored imo" I think the sage brown is more fitting and the green almost looks too much like paint, and same for the opacity, the markings are tattoos, so it would make sense for them to have an almost faded appearance, again all IMO :)

**Review by DatFrenchBanana — CHANGES_REQUESTED** [review]
Tested on MAC 10.13.6. Crusader Kings 3 version 1.3.2(Corvus) with no other mod. Had visual problems for the skin of vanilla ruler designer made characters.

**Comment by zumbak04** [comment]
> Thank you for the detailed explanation. Would it be possible to limit the range that it the marking color may be generated? Color like white, light pink, etc might not just be suitable for markings color.

Don't see it as a problem. Why you don't like white and pink?

> Most of the markings are pretty good given the right color. My next suggestion will be to increase the opacity on them. The base opacity is around half slider in ruler designer:

Changed in 0624e214e56c740796a9afeda76eacd0096d1707.

> One more thing, I found that the dark green haired Nelves have almost black markings, this is pretty unusual for me. It could be a bit more green colored imo:

There are 2 types of blend modes: the first doesn't handle well bright colors (they're too bright and break lighting) and the second doesn't handle well dark colors (the one we use).

**Comment by zumbak04** [comment]
> Tested on MAC 10.13.6. Crusader Kings 3 version 1.3.2(Corvus) with no other mod. Had visual problems for the skin of vanilla ruler designer made characters.

Confirm it still exists on the new CK3 version, please.

**Review by Deuce20201 — APPROVED** [review]
Still works and looks good to me.

**Review by CantTouchThisJinTao — APPROVED** [review]
Tested via ruler designer and didn't notice any issues, markings looked good on varying opacity (except obviously the lower ones where it's barely visible - but I haven't come across any even close to that in-game so no issue there). Also tested in-game (with mixed races as well e.g. NE-Troll/NE-Orc/NE-Gnome/NE-Ogre). Markings look really good, noticed no issues. Tested via giving zealous trait to unmarked females and the facial markings appeared.

Only thing I can really think of is having specific markings relate to specific traits (if that's even possible) e.g. brave = chance of Blades, Bear or Claws etc. But the females in lore can choose which marking they want so it's not really necessary.
Overall I really like these markings and therefore approve considering I personally haven't come across any issues.

**Comment by zumbak04** [comment]
> Only thing I can really think of is having specific markings relate to specific traits (if that's even possible) e.g. brave = chance of Blades, Bear or Claws etc. But the females in lore can choose which marking they want so it's not really necessary.

That's for another PR.

**Review by TheGroxEmpire — COMMENTED** [review]
> Don't see it as a problem. Why you don't like white and pink?

It's just not a usual color for these markings. Combined with already pink / white skin color on some Night Elves, it is bit weird to have those facial marking colors. Although, it is less of an issue now that the opacity has been increased.

- The Tyrande and Azshara facial markings have different color blend mode to the rest:
<details>

image
image
image

image

**Comment by zumbak04** [comment]
> * The Tyrande and Azshara facial markings have different color blend mode to the rest:

Yep, it's intended.

**Review by TheGroxEmpire — APPROVED** [review]
Looks good to me then!

**Review by hamjustine — APPROVED** [review]

**Review by DatFrenchBanana — APPROVED** [review]

**Pull request merged**